### PR TITLE
Ahsu redo confirm claim

### PR DESCRIPTION
**Pull request by ahsu1230**

### CODE REVIEW

File: winterfell/webapp/src/js/claimConfirmCtrl.js [hunk 1]
```diff
@@ -1,8 +1,98 @@
...
+        name: 'Me',
+        email: 'ahsu@gmail.com'
+      },
+      {
+        name: 'Mom',
+        email: 'mom@gmail.com'
+      }
+    ];
+
+    $scope.formsList = [
```
Comment (qwwqwwq) [inline winterfell/webapp/src/js/claimConfirmCtrl.js]: Can you just inject `formTemplateService` instead? It has the list of forms

File: winterfell/webapp/src/js/claimConfirmCtrl.js [hunk 2]
```diff
@@ -1,8 +1,98 @@
...
+app.controller('claimConfirmCtrl', ['$scope', '$state', 'net', 'claimService', '$stateParams', '$uibModal',
+  function($scope, $location, net, claimService, $stateParams, $uibModal) {
     $scope.claimId = $stateParams.claimId;
 
+    $scope.user = {
+      firstName: 'Aaron',
+      lastName: 'Hsu'
+    };
+
+    $scope.vaAddress = {
```
Comment (qwwqwwq) [inline winterfell/webapp/src/js/claimConfirmCtrl.js]: Shouldnt we get these and the following variables from the User model or some other model?

File: winterfell/webapp/src/js/claimConfirmCtrl.js [hunk 3]
```diff
@@ -1,8 +1,98 @@
...
+      return openModifyContactModal('email', $scope.emailList, index);
+    };
+
+    function openModifyContactModal(forType, list, index) {
+      $scope.modalData = {
+        forType: forType,
+        list: list,
+        targetIndex: index >= 0 ? index : -1
+      };
+      return $uibModal.open({
```
Comment (qwwqwwq) [inline winterfell/webapp/src/js/claimConfirmCtrl.js]: Nice!

File: winterfell/webapp/src/js/claimConfirmModifyContactCtrl.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    $scope.targetIndex = modalData.targetIndex;
+    $scope.modalTitle = '';
+    if (modalData.forType == 'address') {
+      $scope.modalTitle = $scope.targetIndex >= 0 ? 'Edit Address' : 'Add Address';
+    } else if (modalData.forType == 'email') {
+      $scope.modalTitle = $scope.targetIndex >= 0 ? 'Edit Email' : 'Add Email';
+    }
+
+    $scope.onCloseModal = function() {
+      $uibModalInstance.close();
```
Comment (qwwqwwq) [inline winterfell/webapp/src/js/claimConfirmModifyContactCtrl.js]: You dont need to make a special function for this, the modal child $scope has certain methods prebound to it by the anguar UI plugin, such that in your modal template you can just say `ng-onclick=close()`

Comment (qwwqwwq) [inline winterfell/webapp/src/js/claimConfirmModifyContactCtrl.js]: Then you can capture a return variable to see if it was accepted, rejected, or dismissed.

File: winterfell/webapp/src/templates/modals/confirmClaimModifyContact.jade [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+  div.input-wrapper(ng-if="forType == 'email'")
+    input(placeholder="Contact Name")
+    input(placeholder="Email")
+    input(placeholder="Confirm Email")
+
+  div.button-wrapper
+    button.cancel-btn(ng-click="onCloseModal()", ng-if="targetIndex < 0") Cancel
+    button.delete-btn(ng-click="onDelete()", ng-if="targetIndex >= 0") Remove
+    button.save-btn(ng-click="onSave()") Save
```
Comment (qwwqwwq) [inline winterfell/webapp/src/templates/modals/confirmClaimModifyContact.jade]: Could do something like `ng-click=close({$value: address})` and catch the return value from uibModal.open and then save that return value to the model in the parent controller.

File: winterfell/webapp/src/templates/modals/confirmClaimModifyContact.jade [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    input.province(placeholder="State/Province")
+    input.zip(placeholder="Zip/Postal")
+
+  div.input-wrapper(ng-if="forType == 'email'")
+    input(placeholder="Contact Name")
+    input(placeholder="Email")
+    input(placeholder="Confirm Email")
+
+  div.button-wrapper
+    button.cancel-btn(ng-click="onCloseModal()", ng-if="targetIndex < 0") Cancel
```
Comment (qwwqwwq) [inline winterfell/webapp/src/templates/modals/confirmClaimModifyContact.jade]: Could do something like `ng-click=close({$value: null})` for this

Comment (ahsu1230) [inline winterfell/webapp/src/templates/modals/confirmClaimModifyContact.jade]: done

File: winterfell/webapp/src/templates/modals/confirmClaimModifyContact.jade (hunk 5 above)
Comment (ahsu1230) [inline winterfell/webapp/src/templates/modals/confirmClaimModifyContact.jade]: done

File: winterfell/webapp/src/js/claimConfirmCtrl.js (hunk 1 above)
Comment (ahsu1230) [inline winterfell/webapp/src/js/claimConfirmCtrl.js]: I'm looking for forms that the user has filled out, not all forms. I don't think `formTemplateService` is what I'm looking for.

File: winterfell/webapp/src/js/claimConfirmCtrl.js (hunk 2 above)
Comment (ahsu1230) [inline winterfell/webapp/src/js/claimConfirmCtrl.js]: done. Was going to do that in a future change (linking actual info instead of placeholder items)

**Comment by ahsu1230** [comment]
Still missing a few links such as:
 * Getting all forms a user filled has completed
 * Getting address of a user
 * Editing physical addresses & emails to send copies to

I want this change to just be ui-only. Separate PR will be for linking things correctly (if they are ready).

**Pull request merged**